// pptxgenjs/index.js
const fs = require("fs");

class Slide {
  constructor() {
    this.elements = [];
  }

  addText(text, options) {
    this.elements.push({ type: "text", text, options });
  }

  addImage(options) {
    this.elements.push({ type: "image", options });
  }

  addShape(shapeType, options) {
    this.elements.push({ type: "shape", shapeType, options });
  }
}

class PptxGenJS {
  constructor() {
    this.author = "";
    this.company = "";
    this.layout = "LAYOUT_WIDE";
    this.slides = [];
    this.ShapeType = PptxGenJS.ShapeType;
  }

  addSlide() {
    const slide = new Slide();
    this.slides.push(slide);
    return slide;
  }

  writeFile({ fileName }) {
    const payload = {
      meta: {
        author: this.author,
        company: this.company,
        layout: this.layout,
      },
      slides: this.slides,
      note: "Arquivo gerado por stub local de PptxGenJS (placeholder).",
    };
    fs.writeFileSync(fileName, JSON.stringify(payload, null, 2));
  }
}

PptxGenJS.ShapeType = {
  rect: "rect",
};

module.exports = PptxGenJS;
